examples/lwt/lwt_echo_server.ml

```diff
@@ -0,0 +1,125 @@
+(* TODO This needs to be paired with the requester example. *)
+(* TODO Usage to comment. *)
+
+let connection_handler : Unix.sockaddr -> Lwt_unix.file_descr -> unit Lwt.t =
+  let module Body = Httpaf.Body in
+  let module Headers = Httpaf.Headers in
+  let module Reqd = Httpaf.Reqd in
+  let module Response = Httpaf.Response in
+  let module Status = Httpaf.Status in
+
+  let request_handler : Unix.sockaddr -> _ Reqd.t -> unit =
+      fun _client_address request_descriptor ->
+
+    let request = Reqd.request request_descriptor in
+    match request.meth with
+    | `POST ->
+      let request_body = Reqd.request_body request_descriptor in
+
+      let response_content_type =
+        match Headers.get request.headers "Content-Type" with
+        | Some request_content_type -> request_content_type
+        | None -> "application/octet-stream"
+      in
+
+      (* Due to a possible bug in http/af, read from the body only once, and
+         create the response based on the data in that first read.
```

Comment:
I think this was fixed in

Reply:
That comment was removed earlier, you may be reviewing an older version of this PR.

Reply:
Yeah I see that now... hmm not sure what's going on.

Reply:
I think I was somehow served a stale version of the PR diff... everything seems to be current now. Sorry about that.